### ISSUE TITLE
Errno 61 Connection refused

### ISSUE THREAD

**Issue by cymerzhang** [Prinsphield/Wechat_AutoJump#99]
你好！我在安装了所有环境，在Xcode里面test也通过的情况下在python3.6 运行。

有如下错误：
(tensorflow) MaggiedeMacBook-Air:Wechat_AutoJump cymer$ python nn_play.py --phone IOS --sensitivity 2.045
/Users/cymer/tensorflow/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: compiletime version 3.5 of module 'tensorflow.python.framework.fast_tensor_util' does not match runtime version 3.6
  return f(*args, **kwds)
2018-01-25 01:07:02.735789: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
WARNING:tensorflow:From nn_play.py:72: all_variables (from tensorflow.python.ops.variables) is deprecated and will be removed after 2017-03-02.
Instructions for updating:
Please use tf.global_variables instead.
==== successfully restored ====
retry to connect, error: HTTPConnectionPool(host='localhost', port=8100): Max retries exceeded with url: /status (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x1154dd978>: Failed to establish a new connection: [Errno 61] Connection refused',))
Traceback (most recent call last):
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/connection.py", line 141, in _new_conn
    (self.host, self.port), self.timeout, **extra_kw)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/util/connection.py", line 83, in create_connection
    raise err
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/util/connection.py", line 73, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/connectionpool.py", line 601, in urlopen
    chunked=chunked)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/connectionpool.py", line 357, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1239, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1285, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1234, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1026, in _send_output
    self.send(msg)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 964, in send
    self.connect()
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/connection.py", line 166, in connect
    conn = self._new_conn()
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/connection.py", line 150, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x1154dd978>: Failed to establish a new connection: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/requests/adapters.py", line 440, in send
    timeout=timeout
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/connectionpool.py", line 639, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/util/retry.py", line 388, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=8100): Max retries exceeded with url: /status (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x1154dd978>: Failed to establish a new connection: [Errno 61] Connection refused',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/wda/__init__.py", line 70, in httpdo
    response = requests.request(method, url, data=data, timeout=HTTP_TIMEOUT)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=8100): Max retries exceeded with url: /status (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x1154dd978>: Failed to establish a new connection: [Errno 61] Connection refused',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/connection.py", line 141, in _new_conn
    (self.host, self.port), self.timeout, **extra_kw)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/util/connection.py", line 83, in create_connection
    raise err
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/util/connection.py", line 73, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/connectionpool.py", line 601, in urlopen
    chunked=chunked)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/connectionpool.py", line 357, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1239, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1285, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1234, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1026, in _send_output
    self.send(msg)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 964, in send
    self.connect()
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/connection.py", line 166, in connect
    conn = self._new_conn()
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/connection.py", line 150, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x1154f50f0>: Failed to establish a new connection: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/requests/adapters.py", line 440, in send
    timeout=timeout
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/connectionpool.py", line 639, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/urllib3/util/retry.py", line 388, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=8100): Max retries exceeded with url: /status (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x1154f50f0>: Failed to establish a new connection: [Errno 61] Connection refused',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "nn_play.py", line 209, in <module>
    AI = WechatAutoJump(args.phone, args.sensitivity, args.serverURL, args.debug, args.resource)
  File "nn_play.py", line 52, in __init__
    self.s = self.client.session()
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/wda/__init__.py", line 252, in session
    sid = self.status()['sessionId']
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/wda/__init__.py", line 191, in status
    res = self.http.get('status')
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/wda/__init__.py", line 101, in fetch
    return self._fetch_no_alert(method, url, data)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/wda/__init__.py", line 107, in _fetch_no_alert
    return httpdo(target_url, method, data)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/wda/__init__.py", line 75, in httpdo
    response = requests.request(method, url, data=data, timeout=HTTP_TIMEOUT)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/Users/cymer/tensorflow/lib/python3.6/site-packages/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=8100): Max retries exceeded with url: /status (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x1154f50f0>: Failed to establish a new connection: [Errno 61] Connection refused',))

请问如何解决？谢谢。

**Comment by leon332157** [comment]
Ip地址和端口正确吗。。。

**Comment by Prinsphield** [comment]
iproxy 8100 8100

**Comment by leon332157** [comment]
@Prinsphield Could you please put it in the README or document, every time people ask about there problems.
能不能放进文档里？

**Comment by Prinsphield** [comment]
I have already added them in both `readme` and `readme_cn`, but many people do not read.

**Comment by leon332157** [comment]
@Prinsphield I think people that are using github are not always developer or even didn't know github, they just download the programs and leave, they don't even look at the README 'cus It's English.
😒

**Comment by Prinsphield** [comment]
i have also provided Chinese readme

**Comment by cymerzhang** [comment]
Thanks guys. That's just the Ip config problem.

@leon332157 , thanks, I am not using github quite often and now a developer that's true. But did read both version Readme, somehow I did not realize I have to change this since in the example it is not mentioned.

@Prinsphield I did read both the English and Chinese version readme. Since it is so nicely documented, I did not realize I have to modified the code. Or you, what might help here is to just add the '--serverURL' argument in your example "Run" example? Either I am really overseeing this sentence or section mentioning this, or it is very difficult to find out.

Sorry for the inconvenience. 谢谢两位。现在程序能跑起来了。
